client/lib/protect-form/index.jsx

```diff
@@ -57,6 +67,41 @@ export const protectForm = WrappedComponent => {
 	};
 };
 
+/*
+ * Declarative variant that takes a 'isChanged' prop.
+ */
+export class ProtectForm extends Component {
+	static propTypes = {
+		isChanged: PropTypes.bool.isRequired
+	}
+
+	// This function binding makes the listener attached to window unique for each
+	// component instance.
+	warnIfChanged = warnIfChanged.bind();
+
+	componentDidMount() {
+		window.addEventListener( 'beforeunload', this.warnIfChanged );
+		if ( this.props.isChanged ) {
+			markChanged( this );
+		}
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'beforeunload', this.warnIfChanged );
```

Comment:
Sorry, 'Feel Free To Ignore'.
I'd noticed that we were removing this listener after making the previous comment on line 80.
In my first pass I hadn't connected the dots... I could have been a bit more clear that it was that comment that could be ignored :)